bdk_esplora_example/src/esplora.rs

```diff
@@ -0,0 +1,186 @@
+                                    loop {
+                                        let new_related_txs = client.scripthash_txs(
+                                            &script,
+                                            Some(related_txs.last().unwrap().txid),
+                                        )?;
+                                        let n = new_related_txs.len();
+                                        related_txs.extend(new_related_txs);
+                                        // we've reached the end
+                                        if n < 25 {
+                                            break;
+                                        }
+                                    }
+                                }
+
+                                Result::<_, esplora_client::Error>::Ok((index, related_txs))
+                            }))
+                        },
+                    )
+                    .collect::<Vec<_>>();
+
+                let n_handles = handles.len();
+
+                for handle in handles {
+                    let (index, related_txs) = handle.join().unwrap()?; // TODO: don't unwrap
+                    if related_txs.is_empty() {
+                        empty_scripts += 1;
+                    } else {
+                        last_active_index = Some(index);
+                        empty_scripts = 0;
+                    }
+                    for tx in related_txs {
+                        let confirmation_time = match tx.status.confirmed {
+                            true => ConfirmationTime::Confirmed {
+                                height: tx.status.block_height.expect("height expected"),
+                                time: tx.status.block_time.expect("blocktime expected"),
+                            },
+                            false => ConfirmationTime::Unconfirmed,
+                        };
+                        if let Err(err) = update.insert_tx(tx.to_tx(), confirmation_time) {
+                            match err {
+                                InsertTxErr::TxTooHigh => {
+                                    /* Don't care about new transactions confirmed while syncing */
+                                }
+                                InsertTxErr::TxMoved => {
+                                    /* This means there is a reorg, we will catch that below */
+                                }
+                            }
+                        }
+                    }
+                }
+
+                if n_handles == 0 || empty_scripts >= stop_gap.unwrap_or(usize::MAX) {
+                    break;
+                }
+            }
+
+            if let Some(last_active_index) = last_active_index {
+                wallet_scan
+                    .last_active_indexes
+                    .insert(keychain, last_active_index);
+            }
+        }
+
+        // This only works on last 10 blocks
```

Comment:
This comment is only true if it's esplora backend. It might be 15 if it's mempool.space. I'd say either delete the comment or make this clear.

Reply:
Noted.. Will update..